src/ts/ContentMetadataBuilder.ts

```diff
@@ -52,47 +35,72 @@ export class ContentMetadataBuilder {
 
-      this.contentMetadata.custom = this.custom;
+      metadata.custom = this.custom;
     }
+    // If the playback has been started, the values cannot be changed and the latest values before the playback started has to be used
+    else {
+      metadata.assetName = this.latestBuiltMetadata.assetName;
+      metadata.viewerId = this.latestBuiltMetadata.viewerId;
+      metadata.streamType = this.latestBuiltMetadata.streamType;
+      metadata.applicationName = this.latestBuiltMetadata.applicationName;
+      metadata.duration = this.latestBuiltMetadata.duration;
+      metadata.custom = this.latestBuiltMetadata.custom;
+    }
+
+    return metadata;
+  }
+
+  private getDynamicMetadata(): Partial<Metadata> {
+    return {
+      encodedFrameRate: this.metadataOverrides.encodedFrameRate || this.metadata.encodedFrameRate,
+      defaultResource: this.metadataOverrides.defaultResource || this.metadata.defaultResource,
+      streamUrl: this.metadataOverrides.streamUrl || this.metadata.streamUrl,
+    };
+  }
+
+  build(): Conviva.ConvivaMetadata {
+    const newMetadata: Partial<Metadata> = {
+      ...this.getStaticMetadata(),
+      ...this.getDynamicMetadata(),
+    };
 
-    this.contentMetadata.encodedFrameRate = this.metadataOverrides.encodedFrameRate || this.metadata.encodedFrameRate;
-    this.contentMetadata.defaultResource = this.metadataOverrides.defaultResource || this.metadata.defaultResource;
-    this.contentMetadata.streamUrl = this.metadataOverrides.streamUrl || this.metadata.streamUrl;
-
-    const convivaContentInfo: ReservedContentMetadata = {
-      [Conviva.Constants.ASSET_NAME]: this.contentMetadata.assetName,
-      [Conviva.Constants.ENCODED_FRAMERATE]: this.contentMetadata.encodedFrameRate,
-      [Conviva.Constants.DURATION]: this.contentMetadata.duration,
-      [Conviva.Constants.DEFAULT_RESOURCE]: this.contentMetadata.defaultResource,
-      [Conviva.Constants.STREAM_URL]: this.contentMetadata.streamUrl,
-      [Conviva.Constants.IS_LIVE]: this.contentMetadata.streamType,
-      [Conviva.Constants.VIEWER_ID]: this.contentMetadata.viewerId || 'GET_VIEWER_ID_FROM_PLAYER',
-      [Conviva.Constants.PLAYER_NAME]: this.contentMetadata.applicationName || 'GET_PLAYER_NAME_OR_TYPE',
+    this.latestBuiltMetadata = newMetadata;
+
+    const newReservedMetadata: ReservedContentMetadata = {
+      [Conviva.Constants.ASSET_NAME]: newMetadata.assetName,
+      [Conviva.Constants.ENCODED_FRAMERATE]: newMetadata.encodedFrameRate,
+      [Conviva.Constants.DURATION]: newMetadata.duration,
+      [Conviva.Constants.DEFAULT_RESOURCE]: newMetadata.defaultResource,
+      [Conviva.Constants.STREAM_URL]: newMetadata.streamUrl,
+      [Conviva.Constants.IS_LIVE]: newMetadata.streamType,
+      [Conviva.Constants.VIEWER_ID]: newMetadata.viewerId || 'GET_VIEWER_ID_FROM_PLAYER',
+      [Conviva.Constants.PLAYER_NAME]: newMetadata.applicationName || 'GET_PLAYER_NAME_OR_TYPE',
```

Comment:
Looks like it was like that already, but matching `applicationName` to `PLAYER_NAME` sounds a bit weird 🤔

Reply:
And here is one more comment https://github.com/bitmovin/bitmovin-player-web-analytics-conviva/pull/106/commits/06cd7394d6fe438d8208460283162123d847bf9f.

Reply:
Thanks for the comments, makes more sense now 🙂